web/models/announcement.ex

```diff
@@ -44,6 +44,12 @@ defmodule Constable.Announcement do
     ]
   end
 
+  def without_interests(query, interests \\ []) do
+    from a in query,
+      full_join: i in assoc(a, :interests),
+      where: not i.name in ^interests or is_nil(i.name)
```

Comment:
Yeah, the query isn't super straightforward, it's basically saying "find all interests that don't have an excluded interest OR it has no interests", because it was filtering announcements that had 0 interests.